app/src/main/java/com/example/tanya/godeltechchallengeandroid/di/module/AppModule.kt

```diff
@@ -0,0 +1,23 @@
+package com.example.tanya.godeltechchallengeandroid.di.module
+
+import android.content.Context
+import com.example.tanya.godeltechchallengeandroid.App
+import dagger.Module
+import dagger.Provides
+import javax.inject.Singleton
+import android.content.SharedPreferences
+import com.example.tanya.godeltechchallengeandroid.util.PREFS_NAME
+
+
+@Module
+class AppModule {
+    @Provides
+    @Singleton
+    fun provideApplication(app : App): Context = app
```

Comment:
The provide method can be changed to the abstract method with the Binds annotation. Please read what the annotation does.